sdk/servicebus/Azure.Messaging.ServiceBus/samples/DeadLetterQueue/Program.cs

```diff
@@ -0,0 +1,301 @@
+                    alias: "--queue",
+                    description: "Service Bus Queue Name to use") { IsRequired = true, Name = "QueueName"},
+                new Option<string>(
+                    alias: "--connection-variable",
+                    description: "The name of an environment variable containing the connection string to use.") { Name = "Connection"},
+            };
+            command.Handler = CommandHandler.Create<string, string, string>(RunAsync);
+            await command.InvokeAsync(args);
+        }
+
+        private static async Task RunAsync(string fullyQualifiedNamespace, string queueName, string connection)
+        {
+            if (!string.IsNullOrEmpty(connection))
+            {
+                _client = new ServiceBusClient(Environment.GetEnvironmentVariable(connection));
+            }
+            else if (!string.IsNullOrEmpty(fullyQualifiedNamespace))
+            {
+                _client = new ServiceBusClient(fullyQualifiedNamespace, new DefaultAzureCredential());
+            }
+            else
+            {
+                throw new ArgumentException(
+                    "Either a fully qualified namespace or a connection string environment variable must be specified.");
+            }
+
+            var cts = new CancellationTokenSource();
+            var sender = _client.CreateSender(queueName);
+
+            // For the delivery count scenario, we first send a single message, 
+            // and then pick it up and abandon it until is "disappears" from the queue.
+            // Then we fetch the message from the dead-letter queue (DLQ) and inspect it. 
+            await SendMessages(sender, 1);
+            await ExceedMaxDeliveryAsync(queueName);
+
+            // For the fix-up scenario, we send a series of messages to a queue, and
+            // run a receive loop that explicitly pushes messages into the DLQ when 
+            // they don't satisfy a processing condition. The fix-up receive loop inspects 
+            // the DLQ, fixes the "faulty" messages, and resubmits them into processing. 
+            var sendTask = SendMessages(sender, int.MaxValue);
+            var receiveTask = ReceiveMessages(queueName, cts.Token);
+            var fixupTask = PickUpAndFixDeadletters(queueName, sender, cts.Token);
+
+            // wait for a key press or 10 seconds
+            await Task.WhenAny(
+                Task.Run(() => Console.ReadKey()),
+                Task.Delay(TimeSpan.FromSeconds(10))
+            );
+
+            // end the processing 
+            cts.Cancel();
+            // await shutdown and exit
+            await Task.WhenAll(sendTask, receiveTask, fixupTask);
+        }
+
+        private static Task SendMessages(ServiceBusSender sender, int maxMessages)
+        {
+            dynamic data = new[]
+            {
+                new {name = "Einstein", firstName = "Albert"}, new {name = "Heisenberg", firstName = "Werner"},
```

Comment:
It may be more clear to keep these to a single item per line, since there's no relationship between the items.  It took me a minute to parse out what was happening.